pandas/core/indexes/base.py

```diff
@@ -229,7 +229,7 @@ class Index(IndexOpsMixin, PandasObject):
     _outer_indexer = libjoin.outer_join_indexer_object
 
     _typ = 'index'
-    _data = None
+    _data = None  # type: Union[np.ndarray, ExtensionArray]
```

Comment:
This, edits in core.accessor, and I think in tseries.frequencies could be separated into a small orthogonal PR